### PR TITLE
Add user registration integration tests

### PR DESCRIPTION
**Pull request by mwtrew**
Adds integration tests for user create/update endpoint.

### CODE REVIEW

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+        HttpServletResponse response = createNiceMock(HttpServletResponse.class);
+        replay(response);
+
+        // Act
+        Response createResponse = usersFacade.createOrUpdateUserSettings(request, response, payload.toString());
+
+        // Assert
+        // check status code was 'OK'
+        Assertions.assertEquals(Response.Status.OK.getStatusCode(), createResponse.getStatus());
+
```
Comment (mlt47) [inline src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java:332]: I guess this doesn't check that the password was actually updated. Is there an easier way of doing this than adding a second step to this test where we change the password back to ERIKA_STUDENT_PASSWORD?

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+    }
+
+    /**
+     * Checks that authenticated requests to update other users of each role by a user with the Teacher role are rejected.
+     *
+     * @throws Exception - not expected.
+     */
+    @ParameterizedTest
+    @MethodSource("allTestUsersProvider")
+    public void createOrUpdateEndpoint_updateAnotherUserWhileLoggedInAsTeacher_failsWithForbidden(RegisteredUserDTO targetUser) throws Exception {
```
Comment (mlt47) [inline src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java:441]: Adding a test where a member of Staff successfully updates a user's details would be good for fuller endpoint coverage.

**Review by mlt47 — CHANGES_REQUESTED** [review]
These are neat tests.

I worry a little about people using the incorrect version of JUnit and then not testing what they think they are testing. For this I am assuming that use of the incorrect version will fail silently, is that correct?
Other than cleaning up the static imports on Integration Test files is there any other way we can minimise the chance of this happening?

File: pom.xml [hunk 3]
```diff
@@ -30,6 +32,8 @@
         <netty.version>4.1.77.Final</netty.version>
         <dependency-check.version>7.1.0</dependency-check.version>
         <ossindex.version>3.2.0</ossindex.version>
+        <surefire.jacoco.args />
+        <failsafe.jacoco.args />
```
[the remark this answers is not in the document]
Comment (mwtrew) [reply pom.xml]: I was getting complaints about these not being declared before use at some point, but I've reverted this and it doesn't seem to be an issue any longer.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/IsaacControllerIT.java [hunk 4]
```diff
@@ -82,14 +83,14 @@ public void searchEndpoint_searchAllTypesSpecificTermAsStudent_returnsOnlyConcep
 
         // Assert
         // check status code is OK
-        assertEquals(Response.Status.OK.getStatusCode(), searchResponse.getStatus());
+        Assertions.assertEquals(Response.Status.OK.getStatusCode(), searchResponse.getStatus());
```
[the remark this answers is not in the document]
Comment (mwtrew) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/IsaacControllerIT.java:86]: >I worry a little about people using the incorrect version of JUnit and then not testing what they think they are testing. For this I am assuming that use of the incorrect version will fail silently, is that correct?
Apparently [it still works](https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4-tips), but I'll make the above changes anyway.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+                        .put("givenName", "test")
+                )
+                .put("userPreferences", new JSONObject())
+                .put("passwordCurrent", JSONObject.NULL);
+
+        HttpServletRequest request = createRequestWithSession();
+        replay(request);
+
+        HttpServletResponse response = createNiceMock(HttpServletResponse.class);
+        replay(response);
```
[the remark this answers is not in the document]
Comment (mwtrew) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java:72]: I don't think so, what happens to them is an implementation detail that I wouldn't consider relevant to the test.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java (hunk 1 above)
Comment (mwtrew) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java:342]: I've just added a check for an updated hash in the DB, which doesn't quite fit with the other tests (which focus only on what the API returns) but is simplest and makes sense for this one IMO.

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mwtrew) [reply pom.xml:36]: > I was getting complaints about these not being declared before use at some point, but I've reverted this and it doesn't seem to be an issue any longer.

I've remembered what this was and reverted the revert, without these IntelliJ's built-in JUnit runner exits immediately for some reason.

File: src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java (hunk 2 above)
Comment (mwtrew) [reply src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java]: Done!

**Pull request merged**